nbdime/tests/conftest.py

```diff
@@ -506,6 +506,7 @@ def _kill_nb_app():
          '--port=%i' % port,
         '--ip=127.0.0.1',
         '--log-level=DEBUG',
+        '--LanguageServerManager.autodetect=False', #' adding this fixed the last failing Python test, that was working locally and simply timing out on CI because search for language servers was taking long time on slow CI file system.
```

Comment:
Would you mind including the name of the test that was failing? This seems like it should be an issue to report upstream, and then we can remove this flag when it is fixed.

Reply:
@krassowski I don't remember exactly which python test was failing when you proposed this fix https://github.com/HaudinFlorence/nbdime/pull/7. Could you please remind me here ? Thanks.

Reply:
I do not see a path forward to fix this yet. Suggestions welcome. There are no tests timing out elsewhere, but in JupyterLab core I added this line to tests to because there is no benefit in searching the filesystem for language serves when not needed https://github.com/jupyterlab/jupyterlab/pull/15061

```
==================================== ERRORS ====================================
____________________ ERROR at setup of test_isgit[notebook] ____________________

    def _kill_nb_app():
        try:
            process.terminate()
        except OSError:
            # already dead
            pass
>       popen_wait(process, 10)

process    = <Popen: returncode: None args: ['/opt/hostedtoolcache/Python/3.11.4/x64/bin/...>

/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/nbdime/tests/conftest.py:494: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/nbdime/tests/conftest.py:29: in popen_wait
    return p.wait(timeout)
        p          = <Popen: returncode: None args: ['/opt/hostedtoolcache/Python/3.11.4/x64/bin/...>
        timeout    = 10
/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/subprocess.py:1264: in wait
    return self._wait(timeout=timeout)
        endtime    = 256.32528873
        self       = <Popen: returncode: None args: ['/opt/hostedtoolcache/Python/3.11.4/x64/bin/...>
        timeout    = 10
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Popen: returncode: None args: ['/opt/hostedtoolcache/Python/3.11.4/x64/bin/...>
timeout = 10

    def _wait(self, timeout):
        """Internal implementation of wait() on POSIX."""
        if self.returncode is not None:
            return self.returncode
    
        if timeout is not None:
            endtime = _time() + timeout
            # Enter a busy loop if we have a timeout.  This busy loop was
            # cribbed from Lib/threading.py in Thread.wait() at r71065.
            delay = 0.0005 # 500 us -> initial delay of 1 ms
            while True:
                if self._waitpid_lock.acquire(False):
                    try:
                        if self.returncode is not None:
                            break  # Another thread waited.
                        (pid, sts) = self._try_wait(os.WNOHANG)
                        assert pid == self.pid or pid == 0
                        if pid == self.pid:
                            self._handle_exitstatus(sts)
                            break
                    finally:
                        self._waitpid_lock.release()
                remaining = self._remaining_time(endtime)
                if remaining <= 0:
>                   raise TimeoutExpired(self.args, timeout)
E                   subprocess.TimeoutExpired: Command '['/opt/hostedtoolcache/Python/3.11.4/x64/bin/python', '-m', 'jupyter_server', '--port=62026', '--ip=127.0.0.1', '--log-level=DEBUG', '--no-browser', '--IdentityProvider.token=nbdime-test-token']' timed out after 10 seconds

delay      = 0.049[130](https://github.com/krassowski/nbdime/actions/runs/6045434096/job/16405481322#step:6:131)42399994083
endtime    = 256.32529052999996
pid        = 0
remaining  = -0.000198[166](https://github.com/krassowski/nbdime/actions/runs/6045434096/job/16405481322#step:6:167)0000[183](https://github.com/krassowski/nbdime/actions/runs/6045434096/job/16405481322#step:6:184)183
self       = <Popen: returncode: None args: ['/opt/hostedtoolcache/Python/3.11.4/x64/bin/...>
sts        = 0
timeout    = 10

/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/subprocess.py:2038: TimeoutExpired
```

Reply:
Ok, let's just leave this one as it is then, but it seems weird to me that the LSP autodetect is a blocking operation...